packages/transactions/src/signatures.ts

```diff
@@ -113,3 +114,19 @@ export function transactionSignature(putativeTransactionSignature: string): Tran
     assertIsTransactionSignature(putativeTransactionSignature);
     return putativeTransactionSignature;
 }
+
+export function assertTransactionIsFullySigned<TTransaction extends Parameters<typeof compileMessage>[0]>(
+    transaction: TTransaction & ITransactionWithSignatures
+): asserts transaction is TTransaction & IFullySignedTransaction {
+    const signerAddressesFromInstructions = transaction.instructions
+        .flatMap(i => i.accounts?.filter(a => isSignerRole(a.role)) ?? [])
+        .map(a => a.address);
+    const requiredSigners = new Set([transaction.feePayer, ...signerAddressesFromInstructions]);
+
+    requiredSigners.forEach(address => {
+        if (!transaction.signatures[address]) {
+            // TODO coded error
+            throw new Error(`Transaction is missing signature for address ${address}`);
```

Comment:
Can I be annoying and ask for this in nice copy-pastable markdown format?
```suggestion
            throw new Error(`Transaction is missing signature for address \`${address}\``);
```